### DISCUSSION TITLE
Possible to get values for other settings?

### DISCUSSION THREAD

**Discussion by CalvinAllen** [VsixCommunity/Community.VisualStudio.Toolkit#343] [Q&A] [answered]
Is it possible - either through the toolkit or otherwise, to get the value for other settings (outside of just our extension itself)? 

Example, we're adding CodeLens features and I'd like to direct users to turn ON CodeLens support if it is currently OFF. I don't want to change the value, just know that its off. Thoughts?
Thanks!

**Comment by madskristensen** [comment, marked as answer]
All settings are stored in the registry, so if there's no API to get the specific setting you're looking for, then look in the registry. Use the [Settings Store Explorer](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.SettingsStoreExplorerPreview) to navigate the VS registry.

**Reply by CalvinAllen** [reply]
Is there an API to query for a setting that is (using the explorer) stored in the "roaming" section? I'm definitely out of my realm here :)

**Reply by madskristensen** [reply]
Yes. You can see how the Settings Store Explorer does it here: https://github.com/pharring/SettingsStoreExplorer/blob/main/src/SettingsStoreExplorer/RoamingSettingsStore.cs

**Reply by CalvinAllen** [reply]
Oh wow, that's awesome - but I do have a follow-up question -

```csharp
[Guid("9B164E40-C3A2-4363-9BC5-EB4039DEF653")]
private static class SVsSettingsPersistenceManager { }
```

I don't quite understand the necessity for that, or exactly what it's doing - though I do realize there seems to be a class with the same name in some Internal VS namespace.
